### PR TITLE
Docs Protobuf Serialization

### PR DESCRIPTION
**Pull request by IamRavikantSingh**
#78
Add inline comments for the wrapper classes of syft-proto.

### CODE REVIEW

File: syftlib/src/main/java/org/openmined/syft/proto/Placeholder.kt [hunk 1]
```diff
@@ -9,17 +9,18 @@ data class Placeholder(
     val description: String
 ) {
     companion object {
+        // Generate Placeholder object by parsing PlaceholderOuterClass.Placeholder object
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/proto/Placeholder.kt:12]: may be also add the class description like 
- Placeholder class is used to reference the order of the tensors in the plan.

File: syftlib/src/main/java/org/openmined/syft/proto/State.kt [hunk 2]
```diff
@@ -9,10 +9,14 @@ data class State(
     val placeholders: List<Placeholder>,
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/proto/State.kt:9]: - State class is responsible for storing all the weights of the neural network. We update these model weights after every `plan.execute`.
- For usage check `SyftModel`

File: syftlib/src/main/java/org/openmined/syft/proto/State.kt [hunk 3]
```diff
@@ -9,10 +9,14 @@ data class State(
...
     val syftTensors: MutableList<SyftTensor>
 ) {
+    // Return a list of TorchTensors from a list of SyftTensors
     fun getTorchTensors() = syftTensors.map { it.getTorchTensor() }
+
+    // Return an array of IValue from the SyftTensors list
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/proto/State.kt:15]: Pytorch Ivalue would be clearer I guess.

File: syftlib/src/main/java/org/openmined/syft/proto/SyftModel.kt [hunk 4]
```diff
@@ -28,13 +28,16 @@ data class SyftModel(
 //        } ?: return null
 //    }
 
+    // Save/Update SyftModel
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/proto/SyftModel.kt:31]: Maybe add a note of how this function is used or just refer to the usage in demo app.

additionally, a more elaborate description would be better like call this function after every gradient step.

File: syftlib/src/main/java/org/openmined/syft/proto/SyftTensor.kt [hunk 5]
```diff
@@ -24,6 +24,7 @@ data class SyftTensor(
     val description: String
 ) {
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/proto/SyftTensor.kt:25]: Similarly, I would like a description of the class `syftensor` as well.

File: syftlib/src/main/java/org/openmined/syft/proto/State.kt (hunk 2 above)
Comment (vkkhare) [reply syftlib/src/main/java/org/openmined/syft/proto/State.kt:11]: lol, By "check SyftModel" I meant add a line of code from SyftModel manipulating state and loading these tensors from the file so that the person reading this could know how it is used and where it is used.

Something like `syftModel.updatemodelparams` from SyftModel

File: syftlib/src/main/java/org/openmined/syft/proto/SyftModel.kt [hunk 6]
```diff
@@ -28,13 +28,19 @@ data class SyftModel(
...
+    // call this function after every gradient step.
+    // e.g: model.updateModel(updatedParams.map { it.toTensor() })
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/proto/SyftModel.kt:33]: perfect!

File: syftlib/src/main/java/org/openmined/syft/proto/State.kt [hunk 7]
```diff
@@ -4,15 +4,22 @@ import org.openmined.syftproto.execution.v1.StateOuterClass
 import org.openmined.syftproto.execution.v1.StateTensorOuterClass
 import org.pytorch.IValue
 
+ // State class is responsible for storing all the weights of the neural network. We update these model weights after every plan.execute.
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/proto/State.kt:7]: Line break needed here.

File: syftlib/src/main/java/org/openmined/syft/proto/SyftModel.kt [hunk 8]
```diff
@@ -7,6 +7,7 @@ import java.io.File
 
 private const val TAG = "SyftModel"
 
+// SyftModel is a model class for neural network weights. kind of like nn.Module.state_dict
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/proto/SyftModel.kt:10]: SyftModel would be a public class available to user. So can you write formal java docs for this class? Like function descriptions and class descriptions using @param, @link etc?

File: syftlib/src/main/java/org/openmined/syft/proto/SyftModel.kt [hunk 9]
```diff
@@ -28,13 +35,27 @@ data class SyftModel(
...
             newModelParams.forEachIndexed { index, pytorchTensor ->
                 state.syftTensors[index] = SyftTensor.fromTorchTensor(pytorchTensor)
             }
         }
     }
+
+    /**
+     * This method is used to load SyftModel from file
+     *
+     * @param modelFile the file containing model state.
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/proto/SyftModel.kt:56]: Better describe file as filepath

File: syftlib/src/main/java/org/openmined/syft/proto/SyftModel.kt [hunk 10]
```diff
@@ -28,13 +35,27 @@ data class SyftModel(
 //        } ?: return null
 //    }
 
+    /**
+     * This method is used to save/update SyftModel class.
+     * call this function after every gradient step.
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/proto/SyftModel.kt:40]: Caps in call. better suited would be 
- This function must be called after every gradient step to update the model state for further plan executions

File: syftlib/src/main/java/org/openmined/syft/proto/SyftModel.kt [hunk 11]
```diff
@@ -7,6 +7,13 @@ import java.io.File
 
 private const val TAG = "SyftModel"
 
+/**
+ * SyftModel is a model class for neural network weights.
+ * kind of like nn.Module.state_dict
```
Comment (vkkhare) [inline syftlib/src/main/java/org/openmined/syft/proto/SyftModel.kt:12]: Instead of kind of like----
Maybe go for something more formal SyftModel is the data model class for storing the weights of the neural network used for training or inference.
Also add the field descriptions for modelName, version, pyGridModelId, etc

**Comment by vkkhare** [comment]
closes #78

**Comment by vkkhare** [comment]
@all-contributors add @IamRavikantSingh for documentation as well.
